MediaBrowser.Common/Plugins/BasePluginOfT.cs

```diff
@@ -153,24 +153,16 @@ public virtual void SaveConfiguration(TConfigurationType config)
             }
         }
 
-        /// <summary>
-        /// Saves the current configuration to the file system.
-        /// </summary>
-        public virtual void SaveConfiguration()
-        {
-            SaveConfiguration(Configuration);
-        }
-
         /// <inheritdoc />
         public virtual void UpdateConfiguration(BasePluginConfiguration configuration)
         {
-            if (configuration == null)
+            Configuration = (TConfigurationType)configuration ?? throw new ArgumentNullException(nameof(configuration));
```

Comment:
Use ArgumentNullException.ThrowIfNull and the cast will not return null but throw an exception